File: app/src/main/java/io/github/ovso/righttoknow/ui/pdfviewer/PDFViewerActivity.java
package io.github.ovso.righttoknow.ui.pdfviewer;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.MenuItem;
import butterknife.BindView;
import com.github.barteksc.pdfviewer.PDFView;
import io.github.ovso.righttoknow.R;
import io.github.ovso.righttoknow.framework.AdsActivity;
import java.io.File;

public class PDFViewerActivity extends AdsActivity {
  @BindView(R.id.pdf_view) PDFView pdfView;

  @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setTitle(R.string.title_child_certified_detail);
    getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    if (getIntent().hasExtra("file")) {
      File file = (File) getIntent().getSerializableExtra("file");
      pdfView.fromFile(file).load();
    }
  }

  @Override protected int getLayoutResId() {
    return R.layout.activity_pdfviewer;
  }

  @Override public boolean onOptionsItemSelected(MenuItem item) {
    finish();
    return true;
  }
}

File: app/src/main/java/io/github/ovso/righttoknow/ui/pdfviewer/PDFViewerActivity.kt
package io.github.ovso.righttoknow.ui.pdfviewer

import android.os.Bundle
import android.view.MenuItem
import io.github.ovso.righttoknow.R
import io.github.ovso.righttoknow.databinding.ActivityPdfviewerBinding
import io.github.ovso.righttoknow.exts.viewBinding
import io.github.ovso.righttoknow.framework.AdsActivity
import kotlinx.android.synthetic.main.content_pdf_viewer.*
import java.io.File

class PDFViewerActivity : AdsActivity() {
  private val binding by viewBinding(ActivityPdfviewerBinding::inflate)

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(binding.root)
    setTitle(R.string.title_child_certified_detail)
    supportActionBar?.setDisplayHomeAsUpEnabled(true)
    if (intent.hasExtra("file")) {
      val file = intent.getSerializableExtra("file") as File
      pdf_view.fromFile(file).load()
    }
    showAd()
  }

  override fun onOptionsItemSelected(item: MenuItem): Boolean {
    finish()
    return true
  }
}
